### PR TITLE
Assignment 1 for review

### PR DESCRIPTION
**Pull request by NayNay71**
Coding submitted for assignment 1 - please review and provide feedback!

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+const double COMMISSION_RATE = .02;
+const double PURCHASE_PRICE = 45.50;
+const double SALE_PRICE = 56.90;
+
+// Variables
+double totalPurchasePrice = SHARES_BOUGHT*PURCHASE_PRICE;
+double totalBuyCommission = totalPurchasePrice * COMMISSION_RATE;
+double totalSalePrice = SHARES_SOLD*SALE_PRICE;
+double totalSaleCommission = totalSalePrice*COMMISSION_RATE;
+double totalProfit = (totalSalePrice-totalSaleCommission)-(totalPurchasePrice+totalBuyCommission);
```
Comment (ninjapanzer) [inline main.cpp:19]: Since these variables are part of our program's scope we should not escalate them to the parent scope.

TL;DR;
I would bring them into main and keep them as close to their use as possible.

Consider this, since constants are immutable there is no real danger of them being accessible in the global scope aside from name collisions since we are not using our own namespace currently.

But the other variables live in a public space and would be accessible anywhere in the program and mutable. This could lead to them being changed in the future without us knowing leading to a bug here.

While this program is not likely to expand the most important job of a developer is to build things can handle change. All software changes all the time. So our job is to control unintended access to the memory defined in our logical operations.

Basically don't provide access to any variables to anything higher than it absolutely needs and if it needs to be higher make sure that access is **READ ONLY**

File: main.cpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+int main() {
+    std::cout<< "Total amount paid for stock was $"
+        <<std::setprecision(7)<<std::showpoint<< totalPurchasePrice<<std::endl;
+    std::cout<< "Total amount of commission paid to broker when stock was bought was $"
+        <<std::setprecision(5)<<std::showpoint<< totalBuyCommission<< std::endl;
```
Comment (ninjapanzer) [inline main.cpp:25]: You should also declare variables as close to usage as possible. In the old days we were told to declare them at the top of the file but this is no longer the case. Now we should do something like this. With variables being somewhat "one and done". Used once and then garbage collected.

Example:

```
// Purchase Info
double totalPurchasePrice = SHARES_BOUGHT*PURCHASE_PRICE;
double totalBuyCommission = totalPurchasePrice * COMMISSION_RATE;
std::cout<< "Total amount paid for stock was $"
     <<std::setprecision(7)<<std::showpoint<< totalPurchasePrice<<std::endl;
std::cout<< "Total amount of commission paid to broker when stock was bought was $"
    <<std::setprecision(5)<<std::showpoint<< totalBuyCommission<< std::endl;

// Sales Info
double totalSalePrice = SHARES_SOLD*SALE_PRICE;
double totalSaleCommission = totalSalePrice*COMMISSION_RATE;
std::cout<<"Total amount of stock sold was $"
     <<std::setprecision(7)<<std::showpoint<< totalSalePrice<< std::endl;
std::cout<<"Total amount of commission paid to broker when stock was sold was $"
     <<std::setprecision(6)<<std::showpoint<< totalSaleCommission<< std::endl;
```

When we get around to functions this organization of our logic makes extracting similar patterns to functions easier.
Already these two sections are starting to look very similar with similar input and output. Just something to stew on for the future.

**Comment by ninjapanzer** [comment]
Love it. :heart:
Spend some time thinking about immutability and how it impacts your code but easy to fix.

File: main.cpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+// Variables
+double totalPurchasePrice = SHARES_BOUGHT*PURCHASE_PRICE;
+double totalBuyCommission = totalPurchasePrice * COMMISSION_RATE;
+double totalSalePrice = SHARES_SOLD*SALE_PRICE;
+double totalSaleCommission = totalSalePrice*COMMISSION_RATE;
+double totalProfit = (totalSalePrice-totalSaleCommission)-(totalPurchasePrice+totalBuyCommission);
+
+int main() {
+    std::cout<< "Total amount paid for stock was $"
```
Comment (froschs) [inline main.cpp:22]: Code looks nice. I personally use a little more spacing to separate the stream insertion operator (<<) for readability.
